src/huts/repositorySpec.js

```diff
@@ -35,487 +36,204 @@ describe("huts repository" , function() {
         });
 
         it("explodes if the database throws an error while checking if the huts table exists", async function() {
-            let db = {
-                query: jasmine.createSpy("db.query").and.callFake(async (query) => {
-                    if(query.startsWith("SELECT 'public.huts'::regclass")) {
-                        throw new Error("exploded while checking if huts table exists");
-                    }
-                })
-            };
-
+            mockFailOnQuery(db, "SELECT 'public.huts'::regclass");
             let hutsRepository = new HutsRepository(db);
 
-            let actualError = null;
-            try {
-                await hutsRepository.initialize();
-            } catch(error) {
-                actualError = error;
-            }
+            let actualError = await getErrorsFromRunningFunction(hutsRepository.initialize);
 
             expect(db.query).toHaveBeenCalledWith("SELECT 'public.huts'::regclass");
             expect(actualError).not.toBe(null);
         });
 
         it("explodes if the database explodes while creating the huts table, if the table does not already exist", async function() {
-            let db = {
-                query: jasmine.createSpy("db.query").and.callFake(async (query) => {
-                    if(query.startsWith("SELECT 'public.huts'::regclass")) {
-                        throw new Error('relation "public.huts" does not exist');
-                    }
-                    if(query.startsWith("CREATE TABLE huts(id uuid UNIQUE PRIMARY KEY, data json NOT NULL)")) {
-                        throw new Error('postgres exploded while creating the huts table');
-                    } else {
-                        throw new Error("Unexpected db call");
-                    }
-                })
-            }
-
-            let actualError = null;
+            mockFailOnQuery(db, [
+                "SELECT 'public.huts'::regclass", 
+                "CREATE TABLE huts(id uuid UNIQUE PRIMARY KEY, data json NOT NULL)"
+            ]);
```

Comment:
I think this is actually wrong. It should be:

```js
mockFailOnQuery(db, {
    "SELECT 'public.huts'::regclass": { message: 'relation "public.huts" does not exist' },
    "CREATE TABLE huts(id uuid UNIQUE PRIMARY KEY, data json NOT NULL)": new Error("postgres exploded while creating huts table")
});
```

The problem is that we want the check for if huts exist ("SELECT 'public.huts'...") to fail specifically in a way that tells us that the db doesn't exist. Right now, it just fails as if something unknown went wrong. The test really shouldn't be passing right now -- it's the wrong kind of failure we are seeing.

Here's how I would start fixing this:

1. Make the test fail by changing the assertions, so the failure in ensuring the huts table is seen as the test failing
2. Change the `mockFailOnQuery` to what I suggest above, and see the test now succeed.

Reply:
i need a little help with this one